coder-sdk/secrets.go

```diff
@@ -19,26 +19,36 @@ type Secret struct {
 // Secrets gets all secrets for the given user
 func (c *Client) Secrets(ctx context.Context, userID string) ([]Secret, error) {
 	var secrets []Secret
-	err := c.requestBody(ctx, http.MethodGet, "/api/users/"+userID+"/secrets", nil, &secrets)
-	return secrets, err
+	if err := c.requestBody(ctx, http.MethodGet, "/api/users/"+userID+"/secrets", nil, &secrets); err != nil {
+		return nil, err
+	}
+	return secrets, nil
 }
 
 // SecretWithValueByName gets the Coder secret with its value by its name.
 func (c *Client) SecretWithValueByName(ctx context.Context, name, userID string) (*Secret, error) {
+	// Lookup the secret from the name.
 	s, err := c.SecretByName(ctx, name, userID)
 	if err != nil {
 		return nil, err
 	}
+	// Pull the secret value.
+	// NOTE: This is racy, but acceptable. If the secret is gone or the permission changed since we looked up the id,
+	//       the call will simply fail and surface the error to the user.
```

Comment:
I don't see this as particularly actionable... if the secret is gone then the CLI will surface a 404 as expected. Changes in permissions would also result in the expected error.

Reply:
Ah, I see. Thanks 👍 .